src/account_facebook/facebook_account_controller.ts

```diff
@@ -234,4 +253,264 @@ export class FacebookAccountController {
+                return "Could not extract profile ID from URL";
+            }
+
+            if (profileId !== this.account?.accountID) {
+                log.error(`FacebookAccountController.verifyFacebookArchive: profile_information.json does not belong to the right account`);
+                return `This archive is for @${profileId}, not @${this.account?.accountID}.`;
+            }
+        } catch {
+            return "Error parsing JSON in profile_information.json";
+        }
+
+        return null;
+    }
+
+    // Return null on success, and a string (error message) on error
+    async importFacebookArchive(archivePath: string, dataType: string): Promise<FacebookImportArchiveResponse> {
+        if (!this.db) {
+            this.initDB();
+        }
+
+        let importCount = 0;
+        const skipCount = 0;
+
+        // If archivePath contains just one folder and no files, update archivePath to point to that inner folder
+        const archiveContents = fs.readdirSync(archivePath);
+        if (archiveContents.length === 1 && fs.lstatSync(path.join(archivePath, archiveContents[0])).isDirectory()) {
+            archivePath = path.join(archivePath, archiveContents[0]);
+        }
+
+        // Load the username
+        let profileId: string;
+
+
+        try {
+            const profileInformationPath = path.join(archivePath, "personal_information/profile_information/profile_information.json");
+            const profileData = JSON.parse(fs.readFileSync(profileInformationPath, 'utf-8'));
+
+            if (!profileData.profile_v2?.profile_uri) {
+                return {
+                    status: "error",
+                    errorMessage: "Could not find profile URI in archive",
+                    importCount: importCount,
+                    skipCount: skipCount,
+                };
+            }
+
+            const profileUrl = profileData.profile_v2.profile_uri;
+            profileId = profileUrl.split('id=')[1] || '';
+
+            if (!profileId) {
+                return {
+                    status: "error",
+                    errorMessage: "Could not extract profile ID from URL",
+                    importCount: importCount,
+                    skipCount: skipCount,
+                };
+            }
+        } catch (e) {
+            return {
+                status: "error",
+                errorMessage: "Error parsing profile information JSON",
+                importCount: importCount,
+                skipCount: skipCount,
+            };
+        }
+
+        // Import posts
+        if (dataType == "posts") {
+            const postsFilenames = await glob(
+                [
+                    // TODO: for really big Facebook archives, are there more files here?
+                    path.join(archivePath, "your_facebook_activity", "posts", "your_posts__check_ins__photos_and_videos_1.json"),
```

Comment:
Probably. We're going to need people with big Facebook accounts to help us test this. I actually discovered the same problem with twitter, that it splits JSON files into multiple files for people with like 100k tweets. We should ask around.